### PR TITLE
Internalize a bunch of JS libraries functions

### PR DESCRIPTION
**Pull request by sbc100**
These functions are not reachable by native code and the
convension is that we prefix such functions with `$` (which means
don't mangle) and use camelCase.

### CODE REVIEW

**Comment by sbc100** [comment]
I believe this kind of change is useful because it help us to reason (as readers of the code) about which functions are in the "native" world and which are not.

File: src/library.js [hunk 1]
```diff
@@ -4520,30 +4520,30 @@ LibraryManager.library = {
     }
   },
 
-  emscripten_builtin_mmap2__deps: ['emscripten_with_builtin_malloc', '$syscallMmap2'],
+  emscripten_builtin_mmap2__deps: ['$withBuiltinMalloc', '$syscallMmap2'],
   emscripten_builtin_mmap2: function (addr, len, prot, flags, fd, off) {
-    return _emscripten_with_builtin_malloc(function () {
+    return withBuiltinMalloc(function () {
```
Comment (tlively) [inline src/library.js:4525]: It's used without the `$`!? No wonder I couldn't get it to work...

Comment (sbc100) [reply src/library.js:4525]: Not sure.. i wanted to confirm so I went and made this wider change.

**Comment by kripken** [comment]
Why did code size change here?

**Comment by sbc100** [comment]
Not exactly sure... but only fastcomp was effected, and things mostly got smaller rather than bigger.  So I didn't investigate.

Also not sure if some of the changes are simply from REBASELINE-ing and there was maybe some changes that were unrelated to this change but were covered by a slop factor?

**Comment by kripken** [comment]
I don't think slop can explain things like this:
```diff
-  "a.js": 5321,
+  "a.js": 5204,
```
That's over 100 bytes which is much bigger than the slop we allow.

This doesn't seem worrying from a code size perspective since as you said things got smaller. But it's weird enough that I think it's worth investigating, as maybe something unexpected changed here that could be a bug?

**Comment by sbc100** [comment]
I did some investigation.  I first rebaselined the test without this change.   Then I run this change and only these two files were effected by this actual change:

```
--- a/tests/code_size/hello_webgl2_fastcomp_asmjs.json
+++ b/tests/code_size/hello_webgl2_fastcomp_asmjs.json
@@ -1,12 +1,12 @@
 {
   "a.html": 580,
   "a.html.gz": 382,
-  "a.js": 5321,
-  "a.js.gz": 2510,
+  "a.js": 5204,
+  "a.js.gz": 2465,
   "a.mem": 321,
   "a.mem.gz": 219,
-  "a.asm.js": 9814,
-  "a.asm.js.gz": 3544,
-  "total": 16036,
-  "total_gz": 6655
+  "a.asm.js": 9750,
+  "a.asm.js.gz": 3517,
+  "total": 15855,
+  "total_gz": 6583
 }
diff --git a/tests/code_size/hello_webgl_fastcomp_asmjs.json b/tests/code_size/hello_webgl_fastcomp_asmjs.json
index 68f3ef2f8..b77173aca 100644
--- a/tests/code_size/hello_webgl_fastcomp_asmjs.json
+++ b/tests/code_size/hello_webgl_fastcomp_asmjs.json
@@ -1,12 +1,12 @@
 {
   "a.html": 580,
   "a.html.gz": 382,
-  "a.js": 4867,
-  "a.js.gz": 2353,
+  "a.js": 4696,
+  "a.js.gz": 2288,
   "a.mem": 321,
   "a.mem.gz": 219,
-  "a.asm.js": 9814,
-  "a.asm.js.gz": 3544,
-  "total": 15582,
-  "total_gz": 6498
+  "a.asm.js": 9750,
+  "a.asm.js.gz": 3517,
+  "total": 15347,
+  "total_gz": 6406
 }
```

So.. it only effect fastcomp and only the asmjs mode of fastcomp.  

I looked at the resulting minified JS output and nothing jumped out.. other than differences in the minified names of things.

I will try to attach the two minified JS files, perhaps you able to see comething different about them (I ran the through clang-format to make them a little more readable).

**Comment by sbc100** [comment]
[js.zip](https://github.com/emscripten-core/emscripten/files/5004510/js.zip)

**Comment by kripken** [comment]
Interesting, thanks... looks like the name changes ended up making closure decide differently on inlining and other things, which affects code size, e.g.
```javascript
  function B(a) {
    a = 2 < a ? p(a) : a;
    return A[a] || document.querySelector(a)
  }

=>

  function A(a) { return 2 < a ? p(a) : a } // not inlined
  function C(a) {
    a = A(a);
    return B[a] || document.querySelector(a)
  }
```
Nothing we should try to do for that.

**Pull request merged**